.github/workflows/pathogen-repo-build.yaml

```diff
@@ -199,6 +197,21 @@ jobs:
           | "$NEXTSTRAIN_GITHUB_DIR"/bin/json-to-envvars
           | tee -a "$GITHUB_ENV"
 
+      - if: inputs.runtime == 'aws-batch'
```

Comment:
Would we want to support AWS roles for builds using docker/conda runtimes that upload/deploy to AWS S3? Or would those still be expected to supply their own credentials?

Reply:
Wait, how would the temp credentials created even be passed to the build runtime?

Reply:
Yes, we would want to make it work for those runtimes on GitHub Actions too. It should be straightforward...

After `configure-aws-credentials`, the temp creds are in the environment as the standard `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and `AWS_SESSION_TOKEN` variables. To be available in the Nextstrain build runtime, they must then be either explicitly passed via `nextstrain build --env ...` or implicitly passed in the absence of any `--env` or `--envdir` option. So the caller of pathogen-repo-build does need to opt-into passing these if it's passing other env vars.

Reply:
Ah okay, so this would require clear documentation for the user to know that they need to pass the `--env` options to interact with AWS in any way.

Reply:
Yep, the same as it is now.

Reply:
Yeah, I guess I mean to document the exact environment variables that need to be passed. I think it's slightly less obvious here because the AWS environment variables are created within the GH Action workflow instead of being user provided secrets.

Reply:
Indeed, that's a good point.

Reply:
I'm going to address runtimes other than AWS Batch in another PR.

I did a quick look with GitHub search and all but two of our relevant workflows specify AWS Batch. The exceptions are [hepB](https://github.com/nextstrain/hepatitisB/blob/7bd1b05e55a7fe0179195d13d44abaf40755d1ef/.github/workflows/run-builds.yaml#L14-L17) and [seasonal-flu's fauna → S3 workflow](https://github.com/nextstrain/seasonal-flu/blob/873467c361738220041fcc0ff230def629c25150/.github/workflows/upload.yaml#L31-L44). They'd indeed both likely want conversion to role assumption in the future.

Reply:
[Added doc](https://github.com/nextstrain/.github/pull/53/files#diff-a708599ac70a96e634c67840937db538813ec036b702104dcf91f6e0e7f0849aR67-R73) on available AWS credentials.

Reply:
Made a checklist item on https://github.com/nextstrain/private/issues/98 for other runtimes.